src/executor/admin/KillQueryExecutor.cpp

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "executor/admin/KillQueryExecutor.h"
+
+#include "context/QueryContext.h"
+#include "planner/plan/Admin.h"
+
+namespace nebula {
+namespace graph {
+folly::Future<Status> KillQueryExecutor::execute() {
+    SCOPED_TIMER(&execTime_);
+    auto *killQuery = asNode<KillQuery>(node());
+    auto inputVar = killQuery->inputVar();
+    auto iter = ectx_->getResult(inputVar).iter();
+    DCHECK(!!iter);
+    QueryExpressionContext ctx(ectx_);
+    auto sessionExpr = killQuery->sessionId();
+    auto epExpr = killQuery->epId();
+
+    // TODO: permision check
+
+    std::vector<meta::cpp2::Session> sessionsInMeta;
+    auto listSessionStatus = qctx()->getMetaClient()->listSessions().via(runner()).thenValue(
+        [&sessionsInMeta](StatusOr<meta::cpp2::ListSessionsResp> resp) {
+            if (!resp.ok()) {
+                return resp.status();
+            }
+            sessionsInMeta = std::move(resp.value()).get_sessions();
+            return Status::OK();
+        }).get();
```

Comment:
Why to use this sync method to block current thread? you can chain all async steps with future::then.